orchestrator/services/processes.py

```diff
@@ -365,29 +378,67 @@ def start_process(
 
     _db_create_process(pstat)
 
-    _safe_logstep_withfunc = partial(_safe_logstep, broadcast_func=broadcast_func)
-    return _run_process_async(pstat.pid, lambda: runwf(pstat, _safe_logstep_withfunc))
+    _safe_logstep_with_func = partial(_safe_logstep, broadcast_func=broadcast_func)
+    return _run_process_async(pstat.pid, lambda: runwf(pstat, _safe_logstep_with_func))
 
 
-def resume_process(
-    process: ProcessTable,
-    *,
+def start_process(
+    workflow_key: str,
     user_inputs: Optional[List[State]] = None,
-    user: Optional[str] = None,
+    user: str = SYSTEM_USER,
     broadcast_func: Optional[BroadcastFunc] = None,
-) -> Tuple[UUID, Future]:
-    """Resume a failed or suspended process.
+) -> UUID:
+    """Start a process for workflow.
 
     Args:
-        process: Process from database
-        user_inputs: Optional user input from forms
-        user: user who resumed this process
+        workflow_key: name of workflow
+        user_inputs: List of form inputs from frontend
+        user: User who starts this process
         broadcast_func: Optional function to broadcast process data
 
     Returns:
         process id
 
     """
+    if app_settings.EXECUTOR == "celery":
```

Comment:
second time: we might be able to improve this by using a Strategy pattern.
https://refactoring.guru/design-patterns/strategy/python/example

Reply:
I like that idea, although I prefer to avoid OOP patterns at almost any cost ;) (OOP patterns fix a problem that shouldn't exist in the first place).

Easiest way is to have dict[str, Callable] where the key can be 'start', 'resume' or 'validate'. I'll spent some time on this to see what I end up with.

Reply:
Changed in last commit